lib/rex/parser/fs/ntfs.rb

```diff
@@ -181,7 +181,8 @@ def cluster_from_attribute_non_resident(attribute, cluster_num = 0, size_max = (
 
             data = ''
             while data.length < size_wanted
-              data << @file_handler.read(size_wanted - data.length)
+              # use an arbitrary block size to avoid target memory consumption
+              data << @file_handler.read([size_wanted - data.length, 2**26].min)
```

Comment:
This arbitrary block size can be optimized.
For example 2**21 allow a 50Mb/s transfert on a network that support it.
More or less slow down the transfer.
I don't know what compromise choose.